src/main/java/ikura/Bot.java

```diff
@@ -0,0 +1,234 @@
+// Bot.java
+// Copyright (c) 2020, zhiayang, Apache License 2.0.
+
+package ikura;
+
+import java.util.Map;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.Optional;
+import java.util.ArrayList;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+import java.util.function.BiConsumer;
+
+import java.util.InputMismatchException;
+import java.util.NoSuchElementException;
+
+import ikura.task.Task;
+import ikura.task.Todo;
+import ikura.task.Event;
+import ikura.task.Deadline;
+
+import ikura.util.Pair;
+import ikura.util.StreamUtils;
+import ikura.util.InvalidInputException;
+
+/**
+ * The main class containing the functionality of the application. Does not own the
+ * tasklist nor the frontend.
+ */
+public class Bot {
+
+    private final TaskList tasks;
+    private final Frontend ui;
+
+    /**
+     * Constructs a new Bot using the given Frontend interface and the given TaskList.
+     *
+     * @param ui    the frontend interface to use.
+     * @param tasks the TaskList to use.
+     */
+    public Bot(Frontend ui, TaskList tasks) {
+        this.tasks  = tasks;
+        this.ui     = ui;
```

Comment:
I believe operators(e.g. =) should be surrounded by 1 space? Seems like this happens at other places too. Could be because I am referring to Java's coding standard. Not sure if Kotlin follows that too.

Reply:
it's not covered in the 2103 guide, but the google style guide they base it on says that horizontal alignment is [acceptable](https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment).

Reply:
Alright then all is good!